lib/authstore/README.md

```diff
@@ -0,0 +1,67 @@
+# AuthStore
+
+> Device access and capability authorization.
+
+## Purpose
+AuthStore is an internal Mapeo Core library used to authorize & verify device access and capabilities through a tree of statements describing each device's role and its relationships to other devices. 
+
+The `AuthStore` class is responsible for authorizing access to Mapeo projects by defining the capabilities of each device through relationships with other devices. The `AuthStore` API is primarily meant to be internal to the higher-level `Mapeo` class API rather than used directly.
+
+When a project is created, `AuthStore` writes records about the project, project creator, and the project creator's device. 
+`AuthStore` is a collection of linked statements about the relationships between devices and the roles & capabilities assigned to each device. The tree of statements starts with the device assigned as project creator, which can then add devices and assign them roles.
+
+By traversing the links between statements about devices and verifying the signatures of statements we can validate each device's capabilities. 
+
+### Available roles & capabilities
+In its current implementation AuthStore has three types of statements that it uses to describe devices: `Device`, `Role`, `CoreOwnership`.
+
+The device statements are used to add, remove, and restore device access. The role statements are used to assign the role and its associated capabilities to a device. The core ownership statements are used to prove that a device owns a specific hypercore.
+
+Each device in a project has a role. Roles define the capabilities a device has in the project. Roles are assigned by other devices. In assigning roles there is a tree of relationships between devices that can be used to validate a device's role & capabilities.
+
+There is a limited set of capabilities that can be assigned to each role:
+- `read` - read data & media in a project
+- `write` - create new data & media in a project
+- `edit` - edit data & media in a project
+- `manage:devices` - assign roles and add, remove, and restore devices in a project
```

Comment:
`manage:devices` could be broken down to be more granular but i'm not sure that it is needed